src/test/java/teammates/test/cases/browsertests/InstructorHomePageUiTest.java

```diff
@@ -222,6 +223,36 @@ private void testCourseLinks() {
 
     }
 
+    private void testDownloadAction() {
+
+        // Test that download result button exist in homePage
+        homePage.verifyDownloadResultButtonExists(feedbackSessionClosed.getCourseId(),
+                feedbackSessionClosed.getSessionName());
+
+        ______TS("Typical case: download report");
+
+        AppUrl reportUrl = createUrl(Const.ActionURIs.INSTRUCTOR_FEEDBACK_RESULTS_DOWNLOAD)
+                .withUserId("CHomeUiT.instructor.tmms")
+                .withCourseId(feedbackSessionClosed.getCourseId())
+                .withSessionName(feedbackSessionClosed.getSessionName());
+
+        homePage.verifyDownloadLink(reportUrl);
+
+        ______TS("Typical case: download report unsuccessfully");
+
+        reportUrl = createUrl(Const.ActionURIs.INSTRUCTOR_FEEDBACK_RESULTS_DOWNLOAD)
+                .withUserId("CHomeUiT.instructor.tmms");
+        browser.driver.get(reportUrl.toAbsoluteString());
+        String afterReportDownloadUrl = browser.driver.getCurrentUrl();
+        assertFalse(reportUrl.toString().equals(afterReportDownloadUrl));
+        // Get an error page due to missing parameters in URL
+        assertTrue("Expected url is InstructorHomePage or Unauthorised page, but is " + afterReportDownloadUrl,
+                afterReportDownloadUrl.contains(Const.ActionURIs.INSTRUCTOR_HOME_PAGE)
+                        || afterReportDownloadUrl.contains(Const.ViewURIs.UNAUTHORIZED));
```

Comment:
Is it not possible to be completely sure which of the following pages will be shown?

Reply:
In `InstructorFeedbackResultsPageUiTest` the instructor can either be instructor or instructor and admin. But for `InstructorHomePageUiTest` the instructor is not an admin, so I think we can be completely sure it is error page. I will make the necessary changes.